src/sequence/mod.rs

```diff
@@ -719,22 +724,368 @@ mod tests {
+
+        }
+
+        #[test]
+        fn proptest_seq_converge_with_many_random_data_across_arbitrary_number_of_replicas(
+            dataset in generate_dataset(500),
+            mut replicas in generate_replicas(50)
+        ) {
+            let dataset_length = dataset.len() as u64;
+
+            // insert our data at replicas
+            for data in dataset {
+                // first generate an op from one replica...
+                let op = replicas[0].append(data)?;
+
+                // then apply this to all replicas
+                for replica in &mut replicas {
+                    replica.apply_data_op(op.clone())?;
+                }
+            }
+
+            verify_data_convergence(replicas, dataset_length);
+
+        }
+
+
+        #[test]
+        fn proptest_converge_with_shuffled_op_set_across_arbitrary_number_of_replicas(
+            dataset in generate_dataset(100),
+            mut replicas in generate_replicas(500)
+        ) {
+            let dataset_length = dataset.len() as u64;
+
+            // generate an ops set from one replica
+            let mut ops = vec![];
+
+            for data in dataset {
+                let op = replicas[0].append(data)?;
+                ops.push(op);
+            }
+
+            // now we randomly shuffle ops and apply at each replica
+            for replica in &mut replicas {
+                let mut ops = ops.clone();
+                ops.shuffle(&mut OsRng);
+
+                for op in ops {
+
+                    replica.apply_data_op(op)?;
+                }
+
+            }
+
+            verify_data_convergence(replicas, dataset_length);
+
+        }
+
+
+
+        #[test]
+        fn proptest_converge_with_shuffled_ops_from_many_replicas_across_arbitrary_number_of_replicas(
+            dataset in generate_dataset(1000),
+            mut replicas in generate_replicas(100)
+        ) {
+            let dataset_length = dataset.len() as u64;
+
+            // generate an ops set using random replica for each data
+            let mut ops = vec![];
+            for data in dataset {
+                if let Some(replica) = replicas.choose_mut(&mut OsRng)
+                {
+                    let op = replica.append(data)?;
+                    ops.push(op);
+                }
+            }
+
+            let opslen = ops.len() as u64;
+            prop_assert_eq!(dataset_length, opslen);
+
+            // now we randomly shuffle ops and apply at each replica
+            for replica in &mut replicas {
+
+                let mut ops = ops.clone();
+                ops.shuffle(&mut OsRng);
+
+                for op in ops {
+                    replica.apply_data_op(op)?;
+                }
             }
+
+            verify_data_convergence(replicas, dataset_length);
+        }
+
+
+
+        #[test]
+        fn proptest_dropped_data_can_be_reapplied_and_we_converge(
+            dataset in prop::collection::vec(generate_seq_entry(), 100..1001)
```

Comment:
What's the reason you want a minimum number of entries here instead of just using the `generate_dataset(1000)` ?

Reply:
in case we dropped everything, this makes that much less unlikely. But maybe isn't needed